lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
1. Yeah, my example is contrived but might be something people are using/would use because we didn't always have the `.symlink()` method.
2. You are totally correct; we need to think through a much better solution that what we are currently doing, but I think that requires my `better-stats` module that tracks more data than we currently get.  Until that time, it's probably best to expect users to handle those edge-cases themselves.
3. The other option would be to remove our `isDirectory` logic from the streams implementation and **only** use it as part of the `default`-er method so users can force junctions themselves.  Thoughts?